x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java

```diff
@@ -186,23 +193,28 @@ String maybeEscape(String value) {
 
         @Override
         boolean needsHeader(RestRequest request) {
-            String header = request.param("header");
+            String header = request.param(URL_PARAM_HEADER);
             if (header == null) {
                 List<String> values = request.getAllHeaderValues("Accept");
                 if (values != null) {
                     // header values are separated by `;` so try breaking it down
                     for (String value : values) {
                         String[] params = Strings.tokenizeToStringArray(value, ";");
                         for (String param : params) {
-                            if (param.toLowerCase(Locale.ROOT).equals("header=absent")) {
+                            if (param.toLowerCase(Locale.ROOT).equals(URL_PARAM_HEADER + "=" + PARAM_HEADER_ABSENT)) {
                                 return false;
                             }
                         }
                     }
                 }
                 return true;
             } else {
-                return !header.toLowerCase(Locale.ROOT).equals("absent");
+                List<String> allowed = List.of(PARAM_HEADER_ABSENT, PARAM_HEADER_PRESENT);
```

Comment:
You could also use Set here, but personally I would use a `switch` statement on the `head.toLowerCase(Locale.ROOT)` to return true, false, and default => throw the exception.

Reply:
Good suggestion, thanks. Fixed.